lib/compute/agent-nodes.ts

```diff
@@ -0,0 +1,185 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import { CfnInstanceProfile, Role, ServicePrincipal } from '@aws-cdk/aws-iam';
+import { Fn, Stack, Tags } from '@aws-cdk/core';
+import { KeyPair } from 'cdk-ec2-key-pair';
+import { InitFile, InitFileOptions } from '@aws-cdk/aws-ec2';
+import { AgentNodeProps } from './jenkins-main-node';
+
+export interface AgentNodeConfig{
+  // eslint-disable-next-line camelcase
+  ami_id : string;
+  // eslint-disable-next-line camelcase
+  ec2_cloud_name : string;
+  // eslint-disable-next-line camelcase
+  instance_type : string;
+  // eslint-disable-next-line camelcase
+  worker_label_string : string;
+  // eslint-disable-next-line camelcase
+  number_of_executors : string;
+  // eslint-disable-next-line camelcase
+  remote_user : string;
+}
+export class AgentNode {
+  public readonly AgentNodeInstanceProfileArn: string;
+
+  public readonly SSHEC2KeySecretId: string;
+
+  constructor(stack: Stack) {
+    const key = new KeyPair(stack, 'AgentNode-KeyPair', {
+      name: 'AgentNodeKeyPair',
+      description: 'KeyPair used by Jenkins Main Node to SSH into Agent Nodes',
+    });
+    Tags.of(key)
+      .add('jenkins:credentials:type', 'sshUserPrivateKey');
+
+    const AgentNodeRole = new Role(stack, 'JenkinsAgentNodeRole', {
+      assumedBy: new ServicePrincipal('ec2.amazonaws.com'),
+    });
+    const AgentNodeInstanceProfile = new CfnInstanceProfile(stack, 'JenkinsAgentNodeInstanceProfile', { roles: [AgentNodeRole.roleName] });
+    this.AgentNodeInstanceProfileArn = AgentNodeInstanceProfile.attrArn.toString();
+    this.SSHEC2KeySecretId = Fn.join('/', ['ec2-ssh-key', key.keyPairName.toString(), 'private']);
+  }
+
+  public asInitFile(filePath: string, stackConfig: AgentNodeProps, config: AgentNodeConfig, options?: InitFileOptions): InitFile {
+    return InitFile.fromString(filePath,
+      `import hudson.model.*
+    import jenkins.model.*
+    import hudson.plugins.ec2.*
+    import com.amazonaws.services.ec2.model.InstanceType
+
+    def instance = Jenkins.getInstance()
+    def init_script = 'sudo mkdir -p /var/jenkins/ && sudo chown -R ec2-user:ec2-user /var/jenkins \
+    && sudo yum install -y java-1.8.0-openjdk cmake python3 python3-pip && sudo yum groupinstall -y "Development Tools" \
+    && sudo ln -sfn \`which pip3\` /usr/bin/pip && pip3 install pipenv && sudo ln -s ~/.local/bin/pipenv /usr/local/bin'
+
+// variables
```

Comment:
Align the comment with the definition, and it isn't very clear to me based on naming the difference between this grouping and the one below it.